### PR TITLE
Move AWS job store uploads/downloads to boto3 (reopened)

### PR DESCRIPTION
**Pull request by w-gao**
## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * Move AWS job store uploads/downloads to boto3

### CODE REVIEW

**Comment by DailyDreaming** [comment]
@w-gao Thanks for reopening this!

**Comment by DailyDreaming** [comment]
@w-gao Is this ready for review?

File: src/toil/jobStores/aws/jobStore.py [hunk 1]
```diff
@@ -407,141 +421,104 @@ def getEmptyFileStoreID(self, jobStoreID=None, cleanup=False, basename=None):
...
+        if os.environ.get('TOIL_S3_USE_SSL', True) == 'False':
+            protocol = 'http'
+        if host:
+            botoargs['endpoint_url'] = f'{protocol}://{host}' + f':{port}' if port else ''
 
-        if botoargs:
-            botoargs['calling_format'] = boto.s3.connection.OrdinaryCallingFormat()
+        # TODO: OrdinaryCallingFormat equivalent in boto3?
+        # if botoargs:
+        #     botoargs['calling_format'] = boto.s3.connection.OrdinaryCallingFormat()
```
Comment (w-gao) [inline src/toil/jobStores/aws/jobStore.py:499]: I wasn't able to find `calling_format` in boto3 and test the endpoint configuration added in #3370, but this should be properly tested.

**Review by w-gao — COMMENTED** [review]
@DailyDreaming I think I addressed most of the review comments in #3153. I’ll leave the replacement of old retries for another ticket since I feel like it would involve breaking the boto3 operations into its own functions in order to use the new retry decorator. 

I noted a few parts that I wasn’t sure about below, but I think this should be ready for review. Thanks again for the thorough review in #3153!

File: src/toil/jobStores/aws/utils.py [hunk 2]
```diff
@@ -365,15 +403,19 @@ def retry_sdb(delays=default_delays, timeout=default_timeout, predicate=retryabl
...
-            or (isinstance(e, ClientError) and 'BucketNotEmpty' in str(e)))
+            # boto3 errors
+            # TODO: switch to @retry decorators
+            or (isinstance(e, ClientError) and
+                ('BucketNotEmpty' in str(e)
+                 # TODO: test these!
+                 or (e.response.get('ResponseMetadata', {}).get('HTTPStatusCode') == 409
+                     and 'try again' in str(e))
+                 or e.response.get('ResponseMetadata', {}).get('HTTPStatusCode') in (
+                     404, 500, 502, 503, 504))))
```
[the remark this answers is not in the document]
Comment (DailyDreaming) [reply src/toil/jobStores/aws/utils.py:418]: These look fine to me.

File: src/toil/jobStores/aws/jobStore.py (hunk 1 above)
Comment (DailyDreaming) [reply src/toil/jobStores/aws/jobStore.py:499]: There isn't one I believe, so we can probably just drop it.  I'm fine with doing that in a subsequent PR and testing myself.  You'd need to download and install minion (which is kind of neat actually).  You're right, it needs proper testing, but that's for a future sprint I think.

**Review by DailyDreaming — APPROVED** [review]
This looks great @w-gao .  Thank you for your quick work on this!

**Pull request merged**